File: Array/MakeTwoArraysEqual.cpp
// Problem: https://leetcode.com/problems/make-two-arrays-equal-by-reversing-sub-arrays/

#include <vector>

using namespace std;

class MakeTwoArraysEqual {
public:
    bool canBeEqual(vector<int>& target, vector<int>& arr) {
        int count[1001];
        memset(count, 0, sizeof(count));        
        for (int elem : target) count[elem]++;
        for (int elem : arr) {
            count[elem]--;
            if (count[elem] < 0) return false;
        }
        return true;
    }
};

File: Array/MakeTwoArraysEqual/go/MakeTwoArraysEqual.go
package _go

// canBeEqual store frequency of each element in map and check if two arrays are equal
// Time Complexity: O(n)
// Space Complexity: O(n)
func canBeEqual(target []int, arr []int) bool {
	m := make(map[int]int)
	for _, v := range target {
		m[v]++
	}
	for _, v := range arr {
		m[v]--
		if m[v] < 0 {
			return false
		}
	}
	return true
}
